x-pack/legacy/plugins/apm/server/routes/create_api/index.ts

```diff
@@ -57,17 +58,26 @@ export function createApi() {
                 >).reduce(
                   (acc, key) => {
                     let codec = params[key];
+                    const value = paramMap[key];
+
                     if (!codec) return acc;
 
                     // Use exact props where possible (only possible for types with props)
                     if ('props' in codec) {
                       codec = t.exact(codec);
                     }
 
-                    const result = codec.decode(paramMap[key]);
+                    // add _debug query parameter to all routes
+                    if (key === 'query') {
+                      codec = t.intersection([codec, debugRt]);
+                    }
```

Comment:
You used to do:

```js
              params: {
                query: params.query
                  ? t.intersection([params.query, debugRt])
                  : debugRt
              },
```

Which made sense to me. But with this change, if the route doesn't define a `query` will _debug still be added?

Reply:
No, that's a good catch actually. The way it's written now none of the {query,path,body} parameters get validated if there's no param types. Let me fix that (and write a test for it). Thanks!